src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -0,0 +1,212 @@
+import { html, LitElement } from "@polymer/lit-element";
+import "@polymer/paper-checkbox/paper-checkbox.js";
+import "@polymer/paper-icon-button/paper-icon-button.js";
+import "@polymer/paper-input/paper-input.js";
+import "@polymer/paper-item/paper-icon-item.js";
+import "@polymer/paper-item/paper-item-body.js";
+
+import "../../../components/ha-card.js";
+
+import { HassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { HomeAssistant } from "../../../types.js";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import { clearItems, fetchItems, addItem, completeItem, saveEdit } from "../../../data/shopping_list";
+
+interface Config extends LovelaceConfig {
+  title?: string;
+}
+
+interface ListItem {
+  id: number;
+  name: string;
+  complete: boolean;
+}
+
+class HuiShoppingListCard extends HassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  protected hass?: HomeAssistant;
+  protected config?: Config;
+  private items?: ListItem[];
+
+  static get properties() {
+    return {
+      hass: {},
+      config: {},
+      items: [],
+    };
+  }
+
+  public getCardSize() {
+    return (this.config ? (this.config.title ? 1 : 0) : 0) + (this.items ? this.items.length : 0);
+  }
+
+  public setConfig(config: Config) {
+    this.config = config;
+    this.items = [];
+    this._fetchData();
+  }
+
+  public connectedCallback() {
+    if (!this.hass) {
+      return;
+    }
+
+    super.connectedCallback();
+    this.hass.connection
+      .subscribeEvents(this._fetchData, "shopping_list_updated")
+      .then(() => this._fetchData());
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+  }
+
+  protected render() {
+    if (!this.config || !this.hass) {
+      return html``;
+    }
+
+    // TODO placeholder='${this.localize('ui.panel.shopping-list.add_item')}' is coming back with an empty value
+    return html`
+      ${this.renderStyle()}
+      <ha-card .header="${this.config.title}">
+        <div class="addRow">
+          <paper-input
+            id='addBox'
+            placeholder='Add Me'
+            @keydown='${(e) => this._addKeyPress(e)}'
+            no-label-float
+          ></paper-input>
+          <paper-icon-button
+            slot="item-icon"
+            icon="hass:plus"
+            @click='${(e) => this._addItem(e)}'
```

Comment:
you can just do `this._addItem`. Lit Element will make sure the call context is set correctly automatically